sdk/clientcore/core/src/main/java/io/clientcore/core/util/union/UnionTypes.java

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.util.union;
+
+/**
+ * This annotation is used to specify the types that are allowed in a {@link Union} type. It allows for better tooling
+ * support to ensure users are properly consuming the types in a {@link Union} type.
+ *
+ * @see Union
+ */
+public @interface UnionTypes {
```

Comment:
Where is this annotation used? Also, should this be in the annotations package with all other annotations?

Reply:
This type is unused, from memory, and was just an idea I had for a future tooling feature. We should remove it for now from this PR.